aiida/backends/control.py

```diff
@@ -24,52 +22,71 @@
 
 MAINTAIN_LOGGER = AIIDA_LOGGER.getChild('maintain')
 
+# Note: these functions are not methods of `AbstractRepositoryBackend` because they need access to the orm.
+# This is because they have to go through all the nodes to gather the list of keys that AiiDA is keeping
+# track of (since they are descentralized in each node entry).
+# See the get_unreferenced_keyset function
+
 
 def repository_maintain(
+    manager: Manager,
```

Comment:
Does this need to be passed in? Why not just always get the manager in the function. That's also what you call in the CLI command. It is a singleton, so it doesn't make sense to me to make it configurable.

Reply:
Mmm ok, I think I need more info about this because maybe this does not work as I was assuming. My intent was:

- In the `storage_maintain` CLI method I am calling the `get_manager()` function because this way I receive the default profile ("default" in this case being used loosely, to mean either the one used in `set default` command or the one passed by the `-p` option). This is the way to indicate target profile while using the CLI.
- In the `repository_maintain` ORM method I am receiving a `Manager` because I thought users could load managers for profiles/backends that are not necessarily the default one and then pass that to this method. Is this not so? The `Manager` class only works under the assumption that the profile/backend it holds is the default one?

Reply:
I think you are confusing the `Manager` with the `Backend`. The `Backend` is a class that is instantiated for a given profile, and it will then know how to connect to the database and repository configured for that profile. You can have multiple of these instances in memory and operate on them at the same time. The `Manager` is a singleton (meaning there should only ever be one instance of it in memory) and it is a simple wrapper around (mainly) the loaded backend. When you call `get_backend` it will instantiate a `Backend` if not already the case and it will _always_ do that for the profile that is currently loaded. So the only way to "specify" the profile there, you simply first load the profile you want and the call `get_backend` on the `Manager`.

Reply:
So you mean like this?

``` python
# Getting `profile` as an input
    from aiida.manage.configuration import load_profile
    from aiida.manage.manager import get_manager

    load_profile(profile=profile.name)
    manager = get_manager()
```

Reply:
Exactly, and the `load_profile` call is done automatically by the CLI, so no need to call it yourself.